### PR TITLE
Fixing key preventing converge by velocity

### PR DESCRIPTION
**Pull request by roigcarlo**
We should probably make a test...

### CODE REVIEW

**Comment by philbucher** [comment]
if this does what I think it does then it is very bad :sweat_smile: :sweat_smile: :sweat_smile:

**Comment by roigcarlo** [comment]
It was doing exactly what you think it was doing :upside_down_face: .

File: applications/FluidDynamicsApplication/custom_strategies/convergence_criteria/vel_pr_criteria.h [hunk 1]
```diff
@@ -141,7 +141,7 @@ class VelPrCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpace >
                         DofValue = itDof->GetSolutionStepValue(0);
                         DofIncr = Dx[DofId];
 
-                        KeyType CurrVar = itDof->GetVariable().Key();
+                        const auto& CurrVar = itDof->GetVariable();
                         if ((CurrVar == VELOCITY_X) || (CurrVar == VELOCITY_Y) || (CurrVar == VELOCITY_Z))
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_strategies/convergence_criteria/vel_pr_criteria.h:145]: @roigcarlo so this check was never `True`?
I mean, how can comparing `KeyType` against `Variable` even compile? Or aren't those the types?

Comment (roigcarlo) [reply applications/FluidDynamicsApplication/custom_strategies/convergence_criteria/vel_pr_criteria.h:145]: Actually I think it was working before. Taking a closer look, it seems that variable_data has the size_t() operator redefined to it return the key also when implicitly converted to a size_t for the comparion with the key (Which was KeyType).

Anyway, probably is less confusing this way.

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/convergence_criteria/vel_pr_criteria.h:145]: :+1: I found it less confusing in this way.

**Comment by rubenzorrilla** [comment]
It seems that the gcc actions are stuck. :confused:

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**